tcms/testruns/views.py

```diff
@@ -38,94 +38,96 @@
-                plan=test_plan,
-                build=form.cleaned_data['build'],
-                manager=form.cleaned_data['manager'],
-                default_tester=default_tester,
-            )
+    def post(self, request):
+        """Display the create test run page."""
 
-            try:
-                assignee_tester = User.objects.get(username=default_tester)
-            except ObjectDoesNotExist:
-                assignee_tester = None
+        # If from_plan does not exist will redirect to plans for select a plan
+        if not request.POST.get('from_plan'):
+            messages.add_message(request,
+                                 messages.ERROR,
+                                 _('Creating a TestRun requires a TestPlan, select one'))
+            return HttpResponseRedirect(reverse('plans-search'))
+
+        plan_id = request.POST.get('from_plan')
+        # case is required by a test run
+        # NOTE: currently this is handled in JavaScript but in the TestRun creation
+        # form cases can be deleted
+        if not request.POST.get('case'):
+            messages.add_message(request,
+                                 messages.ERROR,
+                                 _('Creating a TestRun requires at least one TestCase'))
+            return HttpResponseRedirect(reverse('test_plan_url_short', args=[plan_id]))
+
+        # Ready to write cases to test plan
+        test_cases = get_selected_testcases(request)
+        test_plan = TestPlan.objects.get(plan_id=plan_id)
+
+        # note: ordered by case_id for test_show_create_new_run_page()
+        tcs_values = test_cases.select_related('author',
+                                               'case_status',
+                                               'category',
+                                               'priority').order_by('case_id')
+
+        if request.POST.get('POSTING_TO_CREATE'):
+            form = NewRunForm(request.POST)
+            form.populate(product_id=test_plan.product_id)
+
+            if form.is_valid():
+                # Process the data in form.cleaned_data
+                default_tester = form.cleaned_data['default_tester']
+
+                test_run = TestRun.objects.create(
+                    product_version=test_plan.product_version,
+                    stop_date=None,
+                    summary=form.cleaned_data.get('summary'),
+                    notes=form.cleaned_data.get('notes'),
+                    plan=test_plan,
+                    build=form.cleaned_data['build'],
+                    manager=form.cleaned_data['manager'],
+                    default_tester=default_tester,
+                )
 
-            loop = 1
-            for case in form.cleaned_data['case']:
                 try:
-                    tcp = TestCasePlan.objects.get(plan=test_plan, case=case)
-                    sortkey = tcp.sortkey
+                    assignee_tester = User.objects.get(username=default_tester)
```

Comment:
Remove this entire block around `assignee_tester` and replace this variable with `User.objects.filter(username=default_tester).first()` below (this will reduce number of local variables and complexity)